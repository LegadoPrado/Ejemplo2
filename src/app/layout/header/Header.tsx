import React from "react";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
import {
  getCurrentUser,
  logOut as closeSession,
} from "./../../helpers/auth-helper";
import { Login } from "../../components/login/Login";
import { RegisterForm } from "../../components/registerForm/RegisterForm";
import "./header.scss";

export const Header = () => {
  const user = getCurrentUser();

  const logOut = () => {
    closeSession();
    window.location.href = "./login";
  };

  return (
    <Router>
      <div className="nav">
        <div className="nav-item brand">
          <span>ReactJs</span>
        </div>
        <div className="nav-item menu"></div>
        <div className="nav-item user">
          {!user ? (
            <Link to="/login">
              <span>Login</span>
            </Link>
          ) : (
            <span onClick={logOut}>{user.replaceAll('"', "")}</span>
          )}
        </div>
      </div>
      <Switch>
        <Route path="/register-form">
          <RegisterForm />
        </Route>
        <Route path="/login">
          <Login />
        </Route>
      </Switch>
    </Router>
  );
};
