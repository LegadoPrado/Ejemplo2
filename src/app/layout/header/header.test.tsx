import { shallow } from "enzyme";
import { Header } from "./Header";

describe("Pruebas en header", () => {
  test("Debe de mostrar el componente correctamente ", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});

export {};
