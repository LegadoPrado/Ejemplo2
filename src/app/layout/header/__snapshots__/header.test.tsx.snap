// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en header Debe de mostrar el componente correctamente  1`] = `
<div
  className="nav"
>
  <div />
  <div />
  <div />
</div>
`;
